### ISSUE TITLE
An unregistered block style should not produce a warning

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#3092]
When a custom block extension is not registered, the processor currently issues a warning when it comes across a usage of that block style. For example, this source:

```
[foobar]
--
--
```

produces the following warning when processed without any extensions loaded:

```
asciidoctor: WARNING: test-case.adoc: line 2: invalid style for open block: foobar
```

I believe this discourages writers from using extension blocks in their writing, which is the opposite of the message we want to send. It's also inconsistent with how custom macros are handled, which are left unprocessed.

The built-in behavior should a) be consistent and b) allow the content through without giving it any special meaning. However, we could consider logging a message at the info or debug level to help detect these occurrences when debugging or validating.

If possible, we would still want to warn if the block style is applied to the wrong block context (as that's a misconfiguration).

**Comment by mojavelinux** [comment]
cc: @Mogztter

**Comment by mojavelinux** [comment]
Note that the style is already appended as a role (aka CSS class) in the generated HTML, which helps tools decorate the block in the absence of the extension.

**Comment by Mogztter** [comment]
> I believe this discourages writers from using extension blocks in their writing, which is the opposite of the message we want to send. It's also inconsistent with how custom macros are handled, which are left unprocessed.

:+1: 

> The built-in behavior should a) be consistent and b) allow the content through without giving it any special meaning. However, we could consider logging a message at the info or debug level to help detect these occurrences when debugging or validating.

I wonder if we should do the same for syntax highlighter as well. Because that's basically the same, the content will be here, just "undecorated".

> If possible, we would still want to warn if the block style is applied to the wrong block context (as that's a misconfiguration).

For instance, that would mean that we have a `foobar` block extension registered but not for the `open` context right ?
It's tricky because it could be a misconfiguration or a misuse but yes it should probably be a warning.

**Comment by mojavelinux** [comment]
> I wonder if we should do the same for syntax highlighter as well. Because that's basically the same, the content will be here, just "undecorated".

I think there's a difference between a syntax highlighter than can't be loaded and one that isn't relevant/available in a given environment. If it's not relevant for a given environment, then we could output a debug message. If it fails to load, that would be an warning (as it is).

**Comment by mojavelinux** [comment]
> It's tricky because it could be a misconfiguration or a misuse but yes it should probably be a warning.

Actually, you're right. We can't know what else the style might be used for, so there's no wrong usage.

**Comment by mojavelinux** [comment]
@Mogztter One open question is whether we should log a message in verbose mode if a unregistered block macro is found:

```
foobar::[]
```

Currently, this is just ignored (even in verbose mode).

(I would like to avoid looking for unregistered inline macros because that's a whole other ball field and could introduce a lot of unintended side effects. So in that scenario, unknown syntax is simply ignored).

**Comment by Mogztter** [comment]
> One open question is whether we should log a message in verbose mode if a unregistered block macro is found:

Yes.
In my opinion I think it's important to log a message in verbose mode when we take a decision.
Ignoring or not doing something is a decision.
In this case we've recognized a block macro and we did nothing. I think it's a really useful information and we should let the user know.

> (I would like to avoid looking for unregistered inline macros because that's a whole other ball field and could introduce a lot of unintended side effects. So in that scenario, unknown syntax is simply ignored).

From a user perspective it's the same scenario. But if that could introduce a lot of side effects then we should leave it (at least for now).

**Comment by mojavelinux** [comment]
> In this case we've recognized a block macro and we did nothing.

The reason I brought it up is because we are not currently recognizing a block macro. We only recognize block macros that are known.

But what I could do is check for verbose mode, then look for the block macro. That way, we don't incur any additional cost in normal mode. I'll see if that's feasible.

**Comment by mojavelinux** [comment]
I think I got it.

**Comment by mojavelinux** [comment]
resolved by #3097
